### PR TITLE
[NN] Add EGATConv nn.module

### PR DESCRIPTION
**Pull request by Argusmocny**
## Description

Add EGATConv nn.module on dgl.nn.pytorch
Add unitest for EGATConv 
Add docs for EGATConv

[topic](<https://discuss.dgl.ai/t/new-nn-pytorch-module-gat-layer-with-fully-valuable-edge-features-egat/2388>)

## Checklist
Please feel free to remove inapplicable items for your PR.
- [x] The PR title starts with [$CATEGORY] (such as [NN], [Model], [Doc], [Feature]])
- [x] Changes are complete (i.e. I finished coding on this PR)
- [x] All changes have test coverage
- [x] Code is well-documented
- [x] To the best of my knowledge, examples are either not affected by this change,
      or have been fixed to be compatible with this change (example is available by module docs)
- [x] Related ~~issue~~ discussion is referred in this PR

### CODE REVIEW

**Comment by mufeili** [comment]
@Argusmocny The new implementation looks fine to me in terms of correctness. Great job! Have you tried replacing the previous implementation with the new implementation in your training scripts? It's probably sufficient to train a few iterations or epochs to see if the loss values/evaluation results look similar. If so, we can swap the old implementation with the new implementation.
